doomseeker: fix build by using qt wrapper hook

bhipple opened the pull request:
Current master immediately crashes; this fixes the build and the program runs
successfully. See #65399 for details.

Closed #79128


<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bhipple committed:
doomseeker: fix build by using qt wrapper hook

Current master immediately crashes; this fixes the build and the program runs
successfully. See #65399 for details.

Closed #79128

bhipple commented on pkgs/applications/misc/doomseeker/default.nix:
```diff
@@ -19,9 +19,7 @@ stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
-  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.cc.isClang "-Wno-error=format-security";
```
Is there a way I can test with clang here to see if we should still pass in `stdenv.cc` explicitly just for the `isClang`?  I scanned through the manual but didn't see the clang stdenv documentation.

worldofpeace replied:
Derive it like
```
mkDerivationWith clangStdenv {
```

veprbl replied:
The warning is triggered in macos related code, so it should be enough to s/stdenv.cc.isClang/stdenv.isDarwin/

bhipple replied:
My motivation was wondering if there was some equivalent to `pkgsStatic.doomseeker` that I could try at the cmdline.  My first instinct when seeing these things is to wonder if the opt-out is bitrotted and no longer needed thanks to a version upgrade at some point.  

In this case, it looks like ofborg has confirmed this is still needed.  I'll do it properly by looking at the isClang attr of cc rather than replacing it with isDarwin.

veprbl commented:
@GrahamcOfBorg build doomseeker

worldofpeace commented:
This should be backported to both 19.09 and 20.03 please.

bhipple commented:
@GrahamcOfBorg build doomseeker

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -2877,7 +2877,9 @@ in
 
-  doomseeker = qt5.callPackage ../applications/misc/doomseeker { };
+  doomseeker = qt5.callPackage ../applications/misc/doomseeker {
+    inherit (stdenv.cc) isClang;
```
it would be simpler to just use `stdenv.cc.isClang` in the expression?

bhipple replied:
I'll leave it up to you.  Since this package is supposed to be built with a call to the `mkDerivation inside `qt5`, and is NOT supposed to use `stdenv.mkDerivation`, I think it's a little clearer to leave `stdenv` out of scope entirely and just pass in the boolean, but I don't feel too strongly one way or the other.

veprbl committed:
doomseeker: cleanup

This avoids using NIX_CFLAGS_COMPILE by switching to hardeningDisable.
The hack is also only needed for darwin sources and is not specific to
clang. This also removes non-standard isClang argument.

veprbl commented:
@GrahamcOfBorg build doomseeker

veprbl reviewed (approved):
Tested to run on NixOS, tested to build with the `clangStdenv` on NixOS.

worldofpeace commented:
Thanks for backports @veprbl :sparkles:
